src/parser/GenericParser.ts

```diff
@@ -85,13 +85,15 @@ export class GenericParser {
             for (const file of treeParseResults) {
```

Comment:
Should work. On the other hand, having two `filter` calls on `treeParseResults` should be less efficient than traversing it just once.

But checking for `isErrorFile` should be indeed redundant when already checking for `isParseFile`.

Reply:
what do you mean with two filter calls?

Reply:
The thing were we "add unsupported files later"

Reply:
we need to loop through the results again in line 116 anyways
just add them there?

Reply:
Thats another list (promisesResults vs. treeParseResults). 
But since we introduced `LinesOfCodeRawText`, the unsupported files should also be included in that list. Before that, we filtered them out of that before calling `MetricCalculator`. So yes, we could probably also add them there.

Reply:
Changed that accordingly

Reply:
perfect